pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -102,12 +110,31 @@ func (a Adapter) Create() (err error) {
 			return err
 		}
 		glog.V(3).Infof("Successfully updated component %v", componentName)
+		oldSvc, err := a.Client.KubeClient.CoreV1().Services(a.Client.Namespace).Get(componentName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		serviceSpec.ClusterIP = oldSvc.Spec.ClusterIP
+		objectMetaTemp := objectMeta
+		objectMetaTemp.ResourceVersion = oldSvc.GetResourceVersion()
+
+		_, err = a.Client.UpdateService(objectMetaTemp, *serviceSpec)
+		if err != nil {
+			return err
+		}
+		glog.V(3).Infof("Successfully update Service for component %s", componentName)
 	} else {
 		_, err = a.Client.CreateDeployment(*deploymentSpec)
 		if err != nil {
 			return err
 		}
 		glog.V(3).Infof("Successfully created component %v", componentName)
+		_, err = a.Client.CreateService(objectMeta, *serviceSpec)
```

Comment:
Question about something you brought up before, are we creating only one service for all the components?

Reply:
One service for all ports in devfile.